boot/boot_serial/src/boot_serial.c

```diff
@@ -1289,7 +1289,7 @@ boot_serial_in_dec(char *in, int inlen, char *out, int *out_off, int maxout)
 
 #ifdef __ZEPHYR__
     int err;
-    err = base64_decode( &out[*out_off], maxout - *out_off, &rc, in, inlen - 2);
+    err = base64_decode( &out[*out_off], maxout - *out_off, (size_t *)&rc, in, inlen - 2);
```

Comment:
no... absolutely not, if you get a warning here that's because the size of the variables do not match, and here you are pretending your 4 byte variable is actually 8 bytes, trumping whatever stack memory is around it.

Reply:
But `base64_decode` expects a `size_t * `, and it is done the exact same way few lines below for `__ESPRESSIF__` code. Should we instead change the declaration of `rc` to be `size_t`?

Reply:
yes, it needs to match. If the code below pretends a 4 byte variable is an 8 byte variable then the code below is horribly wrong

Reply:
Thank you, I will rework.